services/ngsf_analysis_service/app.py

```diff
@@ -0,0 +1,296 @@
+       - photometry: the photometry to fit to the model (in csv format)
+       - redshift: the known redshift of the object
+
+    Other analysis services may require additional keys in the `inputs` dictionary.
+    """
+    analysis_parameters = data_dict["inputs"].get("analysis_parameters", {})
+    analysis_parameters = {**default_analysis_parameters, **analysis_parameters}
+
+    fix_z = analysis_parameters.get("fix_z") in [True, "True", "t", "true"]
+
+    # this example analysis service expects the photometry to be in
+    # a csv file (at data_dict["inputs"]["photometry"]) with the following columns
+    # - filter: the name of the bandpass
+    # - mjd: the modified Julian date of the observation
+    # - magsys: the mag system (e.g. ab) of the observations
+    # - flux: the flux of the observation
+    #
+    # the following code transforms these inputs from SkyPortal
+    # to the format expected by ngsfcosmo.
+    #
+
+    rez = {"status": "failure", "message": "", "analysis": {}}
+    try:
+        data = pd.read_csv(io.StringIO(data_dict["inputs"]["spectra"]))
+
+        redshift = Table.read(data_dict["inputs"]["redshift"], format='ascii.csv')
+        z = redshift['redshift'][0]
+    except Exception as e:
+        rez.update(
+            {
+                "status": "failure",
+                "message": f"input data is not in the expected format {e}",
+            }
+        )
+        return rez
+
+    # we will need to write to temp files
+    # locally and then write their contents
+    # to the results dictionary for uploading
+    local_temp_files = []
+
+    SUPERFIT_PATH = 'services/ngsf_analysis_service/NGSF'
+    SUPERFIT_DATA_PATH = f'{SUPERFIT_PATH}/data'
+    SUPERFIT_PARAMETERS_JSON = 'services/ngsf_analysis_service/parameters.json'
+    NGSF = "https://github.com/samanthagoldwasser25/NGSF.git"
+    NGSF_bank = "https://www.wiserep.org/sites/default/files/supyfit_bank.zip"
+    NGSF_zip = f"{SUPERFIT_PATH}/{NGSF_bank.split('/')[-1]}"
+
+    if not os.path.isdir(SUPERFIT_PATH):
+        os.makedirs(SUPERFIT_PATH)
+        git_command = f"git clone {NGSF} {SUPERFIT_PATH}"
+        os.system(git_command)
+        curl_command = f'curl -L -H "Content-Type: application/json" -H "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36" -o {NGSF_zip} {NGSF_bank}'
+        os.system(curl_command)
+
+        with zipfile.ZipFile(NGSF_zip, "r") as z:
+            z.extractall(SUPERFIT_PATH)
+
+    if not os.path.isdir(SUPERFIT_DATA_PATH):
+        os.makedirs(SUPERFIT_DATA_PATH)
+
+    local_temp_files = []
+    plot_data = []
+
+    try:
+        for index, row in data.iterrows():
+            filebase = str(uuid.uuid4())
+            SPECFILE = f'{SUPERFIT_DATA_PATH}/{filebase}.dat'
+            wavelengths = np.array(ast.literal_eval(row['wavelengths']))
+            fluxes = np.array(ast.literal_eval(row['fluxes']))
```

Comment:
This seems like a strange way to read a file. Any idea what is happening here?

Reply:
It's sent as a list (including the []) from the serializer, so its the easiest way to pull this in.